### PR TITLE
Indicate suicide reason with exit code.

### PR DESCRIPTION
**Pull request by jeschkies**
Summary:
The current exit code `137` is confusing since it can indicate an
external `SIGKILL` or a Marathon suicide. This change introduces
the exit code `42` to indicate that Marathon killed itself.

JIRA issues: MARATHON-8381

### CODE REVIEW

File: src/main/scala/mesosphere/marathon/core/base/CrashStrategy.scala [hunk 1]
```diff
@@ -11,6 +11,6 @@ trait CrashStrategy {
 
 case object JvmExitsCrashStrategy extends CrashStrategy {
   override def crash(): Future[Done] = {
-    Runtime.getRuntime.asyncExit()(ExecutionContext.Implicits.global)
+    Runtime.getRuntime.asyncExit(RichRuntime.SuicideExitCode)(ExecutionContext.Implicits.global)
```
Comment (zen-dog) [inline src/main/scala/mesosphere/marathon/core/base/CrashStrategy.scala]: Should we also start differentiating between suicide reasons? E.g. `def crash(reason: SuicideReason)` with e.g. `SuicideOnZKConnectionLoss` etc.?

Comment (jeschkies) [reply src/main/scala/mesosphere/marathon/core/base/CrashStrategy.scala]: I was wondering the same. I'd say yes. However, it requires `def crash()` to take an argument.

Comment (zen-dog) [reply src/main/scala/mesosphere/marathon/core/base/CrashStrategy.scala]: There are 5 usages of it in total and you can easily cover it with a default code.

Comment (kensipe) [reply src/main/scala/mesosphere/marathon/core/base/CrashStrategy.scala]: I'm with the zen-dog!

File: src/main/scala/mesosphere/marathon/core/base/RichRuntime.scala [hunk 2]
```diff
@@ -51,5 +51,6 @@ case class RichRuntime(runtime: Runtime) extends StrictLogging {
 
 object RichRuntime {
   val FatalErrorSignal = 137
+  val SuicideExitCode = 42
```
Comment (zen-dog) [inline src/main/scala/mesosphere/marathon/core/base/RichRuntime.scala]: Those vals deserve "talking names" and comments explaining their usage. wdyt?

File: src/main/scala/mesosphere/marathon/core/base/CrashStrategy.scala (hunk 1 above)
Comment (jeschkies) [reply src/main/scala/mesosphere/marathon/core/base/CrashStrategy.scala]: +1 let's define several exit codes.

We have
* Failed to connect to Zookeeper
* Connection loss to Zookeeper
* Failed to initialize Plugins
* Lost leadership
* Leadership ended with failure
* Leadership ended gracefully
* Error in Mesos scheduler
* Uncaught exception

Am I missing any?

File: src/main/scala/mesosphere/marathon/core/base/RichRuntime.scala (hunk 2 above)
Comment (jeschkies) [reply src/main/scala/mesosphere/marathon/core/base/RichRuntime.scala]: Yup see above.

File: docs/docs/exit-codes.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+| Exit Code | Reason                                                               |
+|-----------|----------------------------------------------------------------------|
+|100        | `ZookeeperConnectionFailure` - Could not connect to Zookeeper        |
+|101        | `ZookeeperConnectionLost` - Lost connect to Zookeeper                |
+|102        | `PluginInitializationFailure` - Could not load plugin                |
+|103        | `LeadershipLoss` - Lost leadership                                   |
+|104        | `LeadershipEndedFaulty` - Leadership ended with an error             |
+|105        | `LeadershipEndedGracefully` - Leadership ended without an error      |
+|106        | `MesosSchedulerErro` - The Mesos scheduler driver had an error       |
```
Comment (ichernetsky) [inline docs/docs/exit-codes.md]: A missing `r` at the end of `MesosSchedulerErro`.

Comment (jeschkies) [reply docs/docs/exit-codes.md]: Thanks!

**Pull request merged**